src/main/java/com/google/devtools/build/lib/skyframe/PackageLookupFunction.java

```diff
@@ -295,17 +295,29 @@ private PackageLookupValue getPackageLookupValue(
   private PackageLookupValue computeWorkspacePackageLookupValue(
       Environment env, ImmutableList<Root> packagePathEntries)
       throws PackageLookupFunctionException, InterruptedException {
-    PackageLookupValue result =
+    PackageLookupValue resultForWorkspaceDotBazel =
+        getPackageLookupValue(
+            env,
+            packagePathEntries,
+            LabelConstants.EXTERNAL_PACKAGE_IDENTIFIER,
+            BuildFileName.WORKSPACE_DOT_BAZEL);
+    if (resultForWorkspaceDotBazel == null) {
+      return null;
+    }
+    if (resultForWorkspaceDotBazel.packageExists()) {
+      return resultForWorkspaceDotBazel;
+    }
```

Comment:
The same logic is repeated again.
Can you extract to a helper function?

Reply:
I cannot refactor this code, because the logic is a bit different from others. It involves checking WORKSPACE file under multiple package paths.